05-deploying_shiny_app.Rmd

```diff
@@ -1,279 +1,239 @@
 
-5. Click 'Begin import'
+You should provide the URL of the app's GitHub repository as well as any existing webapp data sources it should be connected to.
 
+### Clone the repository
 
-![](images/shiny/github_shiny_import.gif)
+1.  Navigate to the app's repository on GitHub.
+2.  Select 'Clone or download'.
+3.  Ensure that the dialogue says 'Clone with SSH'. If the dialogue says 'Clone with HTTPS' select 'Use SSH'.
+4.  Copy the SSH URL.
+5.  In RStudio, select 'File' > 'New project...' > 'Version control' > 'Git'.
+7.  Paste the SSH URL in the 'Repository URL' field.
+8.  Select 'Create Project'.
 
+### Develop the app
 
-### In R Studio in the platform, create a new project from your new Github repository
+Develop the app in RStudio.
 
-In R Studio, navigate through the following menus:
+Your app can take one of several forms:
 
-File -> New project -> Version control -> Git
+1.  A directory containing `server.R`, plus, either `ui.R` or a www directory that contains the file `index.html`.
+2.  A directory containing `app.R`.
+3.  An `.R` file containing a Shiny application, ending with an expression that produces a Shiny app object.
+4.  A list with `ui` and `server` components.
+5.  A Shiny app object created by `shinyApp`.
 
-![](images/shiny/github_shiny_rstudio.gif)
+By default, the template contains `server.R` and `ui.R` files, however, you may wish to take a different approach depending on your requirements. For example, using `app.R`, it is possible to deploy Shiny apps from within a package, as [here](https://github.com/RobinL/costmodelr/blob/b328902026bd1cce5d17b487e310c59725ea4d62/R/shiny_explorer.r#L20).
 
-In 'Repository URL', enter the URL of the repository you just created.  This needs to be the SSH URL (starting with `git@github`), **NOT** the HTTPS url (which starts with `https`). Therefore, it is in the following format:
-
-```
-git@github.com:moj-analytical-services/your_name_goes_here.git
-```
-
-R Studio will create a new project folder with all the template files in.
-
-You can now start building your Shiny app by editing `server.R` and `ui.R` files.
-
-#### Further notes if you're having trouble finding your new repo's url
-
-If you navigate to your new repository's home page (which will have a url in the form `https://github.com/moj-analytical-services/your_name_goes_here`), you can use the following buttons to access this url (make sure you click the 'ssh' button):
-
-![](images/shiny/use_ssh.PNG)
-
-![](images/shiny/ssh_url.PNG)
+### Manage dependencies
 
+Most apps will have dependencies on various third-party packages. Changes to these packages may not always be backwards-compatible. To avoid compatibility issues and ensure reproducible outputs, it is necessary to use a package management system, such as packrat or conda.
```

Comment:
I think it's helpful to describe them with more detail in the first sentence `third-party R packages (e.g. dplyr).`

Reply:
Changed in a741237